docs/migration-guide/to-16.md

```diff
@@ -4,10 +4,53 @@ This release contains significant or breaking changes.
 
 ## Significant changes
 
+- added support for ESM Node.js API and deprecated CJS API
 - added support for ESM plugins
 - added support for ESM custom syntaxes
 - added support for ESM custom formatters
 
+### Added support for ESM Node.js API and deprecated CJS API
+
+You can now use ESM [Node.js API](../user-guide/node-api.md) for Stylelint. For example:
+
+```js
+import stylelint from "stylelint";
+
+const result = await stylelint.lint({
+  code: "a {}",
+  config: { rules: { "block-no-empty": true } }
+});
+
+console.log(result);
+```
+
+> [!WARNING]
+> You can also continue to use CJS API in 16.x, but we will drop the support for CJS in future major versions. We strongly recommend migrating your code using CJS API to ESM.
+
+Currently, Stylelint shows the deprecation warning when using CJS API. To hide the warning, use the [`quietDeprecationWarnings`](../user-guide/options.md#quietdeprecationwarnings) option.
+
+```js
+const stylelint = require("stylelint");
+
+const resultPromise = stylelint.lint({
+  // ...
+  quietDeprecationWarnings: true
+});
+```
+
+#### How to migrate from CJS to ESM
+
+You should also:
+
+- add `"type": "module"` to `package.json`
+- add `"exports": "./index.js"` in `package.json`
+- update the `"engines"` field in `package.json` to Node.js 18: `"node": ">=18"`
+- remove `'use strict';` from all JavaScript files
+- use only full relative file paths for imports: `import x from '.';` → `import x from './index.js';`
+- use the `node:` protocol for Node.js built-in imports
```

Comment:
suggestion
medium

```md
You must:

- add `"type": "module"` to `package.json` if you are not using `.mjs` extension for your files
- use only full relative file paths for imports
   e.g. `import x from '.';` → `import x from './index.js';`

You may:

- have to update the `"engines"` field in `package.json` to Node.js 18: `"node": ">=18.12.0"`
- remove `'use strict';` from all JavaScript files
- add `"exports": "./index.js"` in `package.json`
- use the `node:` protocol for Node.js built-in imports
```

notice in particular this change : ">=18.0.0" to ">=18.12.0" (match ours)

Reply:
Sounds good. Updated via ef77af64c.